File: routes/create.js
const express = require('express');
const router = express.Router();
const config = require('../config');
const rds = require('../rds');

router.get('/', function(req, res, next) {
    const [pool, rdsUrl] = rds();
    pool.getConnection(function(err, con) {
        if (err) {
            return next(err);
        }
        const queries = [
            'CREATE DATABASE IF NOT EXISTS hotel;',
            'USE hotel;',
            'CREATE TABLE IF NOT EXISTS rooms(id int NOT NULL, floor int, hasView boolean, occupied boolean, comment varchar(60), PRIMARY KEY(id));'
        ];

        const executeQueries = (index) => {
            if (index === queries.length) {
                con.release();
                return res.render('create', { menuTitle: config.app.hotel_name, url: rdsUrl });
            }
            con.query(queries[index], function(error, result, fields) {
                if (error) {
                    con.release();
                    return next(error);
                }
                executeQueries(index + 1);
            });
        };

        executeQueries(0);
    });
});

module.exports = router;

File: routes/create.py
from flask import Blueprint, render_template
from database import rds

bp = Blueprint('create', __name__)


@bp.route('/create')
def create():
    pool, rds_url = rds()
    try:
        connection = pool.get_connection()
        cursor = connection.cursor()
        cursor.execute('CREATE DATABASE IF NOT EXISTS hotel;')
        cursor.execute('USE hotel;')
        cursor.execute('CREATE TABLE IF NOT EXISTS rooms(id int NOT NULL, floor int, hasView boolean, occupied boolean, comment varchar(60), PRIMARY KEY(id));')
        connection.commit()
        cursor.close()
        connection.close()
        print("Create table in database if not exists!")
    except Exception as e:
        print(f"Error creating database or table: {e}")

    return render_template('create.html', menuTitle=config['app']['hotel_name'], url=rds_url)